api/v1alpha1/orchestrator_types.go

```diff
@@ -20,149 +20,182 @@ import (
-	Enabled             bool         `json:"enabled,omitempty"`
-	EnableGuestProvider bool         `json:"enableGuestProvider,omitempty"`
-	CatalogBranch       string       `json:"catalogBranch,omitempty"`
-	Subscription        Subscription `json:"subscription,omitempty"`
-	SecretRef           SecretRefBS  `json:"secretRef,omitempty"`
+	// Determines whether the RHDH operator should be installed
+	// This determines the deployment of the RHDH instance.
+	// Defaults to false
+	// +kubebuilder:default=false
+	InstallOperator bool `json:"installOperator,omitempty"`
+
+	// Determines whether to enable the guest provider in RHDH.
+	// This should be used for development purposes ONLY and should not be enabled in production.
+	// Defaults to false.
+	// +kubebuilder:default=false
+	DevMode bool `json:"devMode,omitempty"`
+
+	// Configuration for RHDH Plugins.
+	RHDHPlugins RHDHPlugins `json:"plugins,omitempty"`
 }
 
-type PluginDetails struct {
-	Package   string `json:"package,omitempty"`
-	Integrity string `json:"integrity,omitempty"`
+type RHDHPlugins struct {
+	// Notification email plugin configuration
+	NotificationsConfig NotificationConfig `json:"notificationsEmail,omitempty"`
 }
 
 type NotificationConfig struct {
-	Enabled   bool   `json:"enabled,omitempty"`
-	Port      int    `json:"port,omitempty"`
-	Sender    string `json:"sender,omitempty"`
-	Recipient string `json:"replyTo,omitempty"`
-}
+	// Determines whether to install the Notifications Email plugin
+	// Requires setting the hostname and credentials in backstage secret
+	// The secret backstage-backend-auth-secret is created as pre-requisite
+	// See plugin configuration at https://github.com/backstage/backstage/blob/master/plugins/notifications-backend-module-email/config.d.ts
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
 
-type RHDHPlugins struct {
-	NpmRegistry         string                   `json:"npmRegistry,omitempty"`
-	Scope               string                   `json:"scope,omitempty"`
-	Plugins             map[string]PluginDetails `json:"plugins,omitempty"`
-	NotificationsConfig NotificationConfig       `json:"notificationsConfig,omitempty"`
+	// SMTP server port
+	// +kubebuilder:default=587
+	Port int `json:"port,omitempty"`
+
+	// Email address of the Sender
+	// +kubebuilder:default=""
+	Sender string `json:"sender,omitempty"`
+
+	// Email address of the Recipient
+	// +kubebuilder:default=""
+	Recipient string `json:"replyTo,omitempty"`
 }
 
 type Postgres struct {
-	ServiceName      string             `json:"serviceName,omitempty"`
-	ServiceNameSpace string             `json:"serviceNamespace,omitempty"`
-	AuthSecret       PostgresAuthSecret `json:"authSecret,omitempty"`
-	DatabaseName     string             `json:"database,omitempty"`
+	// Name of the Postgres DB service to be used by platform services
+	// +kubebuilder:validation:Required
+	ServiceName string `json:"serviceName"`
```

Comment:
why  `ServiceName` and not just `Name`?

Reply:
Renamed to `name` and `namespace`, also on the crd level